ntoskrnl/ex/profile.c

```diff
@@ -138,7 +139,12 @@ NtCreateProfile(OUT PHANDLE ProfileHandle,
     }
 
     /* Make sure that the buckets can map the range */
-    if ((RangeSize >> (BucketSize - 2)) > BufferSize)
+    BucketsRequired = RangeSize >> BucketSize;
```

Comment:
No idea whether we could add in the comments, that this fix comes straightforwardly from
https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ex/profile/bugdemo.htm
😂

Reply:
Actually, it doesn't -- his fix is buggy :)

Reply:
you mean, because he increases `BucketsRequired` only when the `RangeSize & ...` is `== 0` instead?

Reply:
Yep, it should be != 0

Reply:
Actually, what do you think about
```
BucketsRequired = ALIGN_UP_BY(RangeSize, 1 << BucketSize) >> BucketSize;
```
in terms of readability?

Reply:
It would be better I think; thanks to the macro name we understand what's being done (& the intent).